PhysicsTools/PatAlgos/plugins/PATMuonProducer.cc

```diff
@@ -468,7 +506,57 @@ void PATMuonProducer::produce(edm::Event & iEvent, const edm::EventSetup & iSetu
               } 
           }
       }
-
+      // MC info
+      aMuon.initSimInfo();
+      if (simInfoIsAvailalbe){
+	switch ((*simClassification)[muonBaseRef]){
+	case 0: 
+	  aMuon.setSimType(pat::Muon::NotMatched);
+	  break;
+	case 1: 
+	  aMuon.setSimType(pat::Muon::MatchedPunchthrough);
+	  break;
+	case 11: 
+	  aMuon.setSimType(pat::Muon::MatchedElectron);
+	  break;
+	case 4: 
+	  aMuon.setSimType(pat::Muon::MatchedPrimary);
+	  break;
+	case 3: 
+	  aMuon.setSimType(pat::Muon::MatchedHeavyFlavour);
+	  break;
+	case 2: 
+	  aMuon.setSimType(pat::Muon::MatchedLightFlavour);
+	  break;
+	case -1: 
+	  aMuon.setSimType(pat::Muon::GhostPunchthrough);
+	  break;
+	case -11: 
+	  aMuon.setSimType(pat::Muon::GhostElectron);
+	  break;
+	case -4: 
+	  aMuon.setSimType(pat::Muon::GhostPrimary);
+	  break;
+	case -3: 
+	  aMuon.setSimType(pat::Muon::GhostHeavyFlavour);
+	  break;
+	case -2: 
+	  aMuon.setSimType(pat::Muon::GhostLightFlavour);
+	  break;
+	default:
+	  throw cms::Exception("Error") << "uknown MuonSimHitClassification type";
```

Comment:
LogError here seems more appropriate.
For single release self-consistency it would help to make simClassification to be of the same enum type as MuonSimType.
If the set of types will need to evolve, reading old versions can become problematic if the exception here is kept.
Also, if the association type is specific to some workflow or type of MC, it may not be possible to detect it until production jobs start running and this variable doesn't seem important enough to stop the processing.

Reply:
> LogError here seems more appropriate.

Why? It's an exception. Why LogError is better?

Reply:
I provided an explanation already.

Reply:
fixed